lightning/src/ln/msgs.rs

```diff
@@ -2739,13 +2743,18 @@ impl<'a> Writeable for OutboundOnionPayload<'a> {
 			},
 			Self::Receive {
 				ref payment_data, ref payment_metadata, ref keysend_preimage, sender_intended_htlc_amt_msat,
-				cltv_expiry_height, ref custom_tlvs,
+				cltv_expiry_height, ref sender_custom_tlvs, user_custom_tlvs
 			} => {
 				// We need to update [`ln::outbound_payment::RecipientOnionFields::with_custom_tlvs`]
 				// to reject any reserved types in the experimental range if new ones are ever
 				// standardized.
+				let user_custom_tlv = (!user_custom_tlvs.is_empty()).then(|| (65537, user_custom_tlvs.to_vec()));
```

Comment:
Why do we need to allocate a new `Vec` here?

Reply:
After discussion offline, we concluded that a vec allocation is necessary here.